decayAmplitude/massDependence.cc

```diff
@@ -564,6 +617,231 @@ piPiSWaveAuMorganPenningtonKachaev::piPiSWaveAuMorganPenningtonKachaev()
+	  _piChargedMass(0.0),
+	  _kaonChargedMass(0.0)
+{
+	particleDataTable& pdt = particleDataTable::instance();
+	const string partList[] = { "pi+", "K+" };
+	for (unsigned int i = 0; i < sizeof(partList) / sizeof(partList[0]); ++i)
+		if (not pdt.isInTable(partList[i])) {
+			printErr<< "cannot find particle " << partList[i] << " in particle data table. "
+			<< "Aborting..." << endl;
+			throw;
+		}
+	_piChargedMass = pdt.entry("pi+")->mass();
+	_kaonChargedMass = pdt.entry("K+")->mass();
+}
+
+
+
+boost::shared_ptr<KPiSGLASS>
+KPiSGLASS::Create(const libconfig::Setting* massDepKey){
+	if (massDepKey != nullptr) {
+		double a, r, M0, G0, phiF, phiR, phiRsin, F, R, MMax;
+		string tag;
+		if(not massDepKey->lookupValue("a", a)){
+			printErr << "cannot find 'a' value in configuration for mass dependence '" << Name() << "'. Aborting..." << endl;
+			throw;
+		}
+
+		if(not massDepKey->lookupValue("r", r)){
+			printErr << "cannot find 'r' value in configuration for mass dependence '" << Name() << "'. Aborting..." << endl;
+			throw;
+		}
+
+		if(not massDepKey->lookupValue("M0", M0)){
+			printErr << "cannot find 'M0' value in configuration for mass dependence '" << Name() << "'. Aborting..." << endl;
+			throw;
+		}
+
+		if(not massDepKey->lookupValue("G0", G0)){
+			printErr << "cannot find 'G0' value in configuration for mass dependence '" << Name() << "'. Aborting..." << endl;
+			throw;
+		}
+
+		if(not massDepKey->lookupValue("phiF", phiF)){
+			printErr << "cannot find 'phiF' value in configuration for mass dependence '" << Name() << "'. Aborting..." << endl;
+			throw;
+		}
+
+		if(not massDepKey->lookupValue("phiR", phiR)){
+			printErr << "cannot find 'phiR' value in configuration for mass dependence '" << Name() << "'. Aborting..." << endl;
+			throw;
+		}
+
+		if(not massDepKey->lookupValue("phiRsin", phiRsin)){
+			printErr << "cannot find 'phiRsin' value in configuration for mass dependence '" << Name() << "'. Aborting..." << endl;
+			throw;
+		}
+
+		if(not massDepKey->lookupValue("F", F)){
+			printErr << "cannot find 'F' value in configuration for mass dependence '" << Name() << "'. Aborting..." << endl;
+			throw;
+		}
+
+		if(not massDepKey->lookupValue("R", R)){
+			printErr << "cannot find 'R' value in configuration for mass dependence '" << Name() << "'. Aborting..." << endl;
+			throw;
+		}
+
+		if(not massDepKey->lookupValue("MMax", MMax)){
+			printErr << "cannot find 'MMax' value in configuration for mass dependence '" << Name() << "'. Aborting..." << endl;
+			throw;
+		}
+
+		if(not massDepKey->lookupValue("tag", tag)){
+			printErr << "cannot find 'tag' value in configuration for mass dependence '" << Name() << "'. Aborting..." << endl;
+			throw;
+		}
+
+		return KPiSGLASS::Create(a, r, M0, G0, phiF, phiR, phiRsin, F, R, MMax, tag);
+	} else {
+		printWarn << "no configuration given for mass dependence '" << Name() << "'. Using dummy config" << endl;
+		return KPiSGLASS::Create(0, 0, 0, 0, 0, 0, 0, 0, 0, 0, "");
```

Comment:
Are there meaningful default parameter values?

Reply:
The only reasonable default parameters would the the ones of the simplifies version called LASS, which resulted from a fit to LASS data.
However, there are also more results flying around:
- The original one form the LASS paper
- A refit by Dunwoodie (just some webpage), whose  parameters were used in a recent LHCb analysis
So I would leave it to the user to choose some parameters.